src/main/java/com/iyzipay/AbstractHttpClient.java

```diff
@@ -0,0 +1,130 @@
+package com.iyzipay;
+
+import com.google.gson.Gson;
+import com.iyzipay.exception.HttpClientException;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.methods.*;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.util.EntityUtils;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.List;
+import org.apache.http.client.config.RequestConfig;
+
+abstract class AbstractHttpClient implements HttpClient {
```

Comment:
I guess this is irrelevant. Just add an HttpClient interface, and add your own logic in your code.